addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/TeslaBindingConstants.java

```diff
@@ -0,0 +1,1016 @@
+				if (s.equals("true") || s.equals("1")) {
+					return super.getState("ON");
+				}
+				if (s.equals("false") || s.equals("0")) {
+					return super.getState("OFF");
+				}
+				return super.getState(s);
+			}
+		},
+		SUN_ROOF_STATE("sun_roof_state", "sunroofstate", StringType.class,
+				false), SUN_ROOF("sun_roof_percent_open", "sunroof",
+				PercentType.class, false), TEMPERATURE(null, "temperature",
+				DecimalType.class, false), TIME_TO_FULL_CHARGE(
+				"time_to_full_charge", "timetofullcharge", DecimalType.class,
+				false), TIMESTAMP("timestamp", "eventstamp",
+				DateTimeType.class, false) {
+			@Override
+			public State getState(String s) {
+				Date date = new Date();
+				SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat(
+						"yyyy-MM-dd'T'HH:mm:ss");
+
+				date.setTime(Long.valueOf(s));
+				return super.getState(DATE_FORMATTER.format(date));
+			}
+		},
+		TRIP_CARGING("trip_charging", "tripcharging", OnOffType.class, false) {
+			@Override
+			public State getState(String s) {
+				if (s.equals("true") || s.equals("1")) {
+					return super.getState("ON");
+				}
+				if (s.equals("false") || s.equals("0")) {
+					return super.getState("OFF");
+				}
+				return super.getState(s);
+			}
+		},
+		USABLE_BATTERY_LEVEL("usable_battery_level", "usablebatterylevel",
+				DecimalType.class, false), USER_CHARGE_ENABLE_REQUEST(
+				"user_charge_enable_request", "userchargeenablerequest",
+				StringType.class, false), VEHICLE_NAME("vehicle_name", "name",
+				StringType.class, true) {
+			@Override
+			public State getState(String s) {
+				return super.getState(s.replaceAll("\"", ""));
+			}
+		},
+		VALET_MODE("valet_mode", "valetmode", OnOffType.class, false) {
+			@Override
+			public State getState(String s) {
+				if (s.equals("true") || s.equals("1")) {
+					return super.getState("ON");
+				}
+				if (s.equals("false") || s.equals("0")) {
+					return super.getState("OFF");
+				}
+				return super.getState(s);
+			}
+		},
+		VALET_PIN("valet_pin_needed", "valetpin", OnOffType.class, false) {
+			@Override
+			public State getState(String s) {
+				if (s.equals("true") || s.equals("1")) {
+					return super.getState("ON");
+				}
+				if (s.equals("false") || s.equals("0")) {
+					return super.getState("OFF");
+				}
+				return super.getState(s);
+			}
+		},
+		VIN("vin", "vin", StringType.class, true), WAKEUP(null, "wakeup",
+				OnOffType.class, false) {
+			@Override
+			public State getState(String s) {
+				if (s.equals("true") || s.equals("1")) {
+					return super.getState("ON");
+				}
+				if (s.equals("false") || s.equals("0")) {
+					return super.getState("OFF");
+				}
+				return super.getState(s);
+			}
+		},
+		WHEEL_TYPE("wheel_type", "wheeltype", StringType.class, true);
+
+		private final String RESTID;
+		private final String channelID;
+		private Class<? extends Type> typeClass;
+		private final boolean isProperty;
+
+		public static boolean metric = false;
```

Comment:
Tesla is adding things with every release of the software, and nothing is documented. Currently everything is in here, but some stuff I do not know what it exactly embodies. Miles it is, but see for example like 464 as how I deal with this